### ISSUE TITLE
Markdown render error

### ISSUE THREAD

**Issue by xf-** [codimd/server#402]
In this special case with indent afterwards will case a rendering issue with checkboxes.

### Demo
https://demo.codimd.org/BNKA9JAnTb28a1nOlgMJ7Q#

### Content
```markdown
## Header
- [ ] test test test 
- [ ] test test test
    - test `te` st `test` a
    - test `te` st `test` a
- [ ] `foo` test test `bar` boo
    - `test` asdasdasd http://test.test/foo.html
    - `test` asdas http://test.test/foo.html


    `test`
```

### Screenshot
![image](https://user-images.githubusercontent.com/1295633/84576623-7d88ec80-adb6-11ea-8cff-600030a64207.png)

**Comment by mrdrogdrog** [comment]
This problem occurs only in 1.6.x.
The updated check list lib in 2.0 handles the code correctly.

**Comment by xf-** [comment]
v2 has no stable release and can't remember the last opened documents. As far i understand master is still on v1 until 1.6.1 is released. If a fix is possible (without a rewrite) would be nice.

**Comment by mrdrogdrog** [comment]
That's right. CodiMD 1.6.1 uses its own implementation of the check list rendering. 2.0 will use a lib for that. But 2.0 is far from finished. I don't think that a fix is possible without reworking the rendering code :/

**Comment by SISheogorath** [comment]
Let's see if we can fix it.

The implementation is located over here:
https://github.com/codimd/server/blob/33150b79c7a4c3ba456afdadb31c709b3be18980/public/js/extra.js#L270-L278

If someone wants to tinker around, feel free to try :)

**Comment by davidmehren** [comment]
I narrowed it down a bit with the following even more minimal example:
```
- [ ] test

  `indented code`
```
When `indented code` is indented with less than two spaces, everything is fine. The [generated HTML from MarkdownIt](https://github.com/codimd/server/blob/master/public/js/index.js#L2746) looks like this:
```
<h2 class=\"raw part\" data-startline=\"1\" data-endline=\"1\">Header</h2>
<ul class=\"part\" data-startline=\"2\" data-endline=\"3\">
<li class=\"raw\" data-startline=\"2\" data-endline=\"3\">[ ] test</li>
</ul>
<p class=\"part\" data-startline=\"4\" data-endline=\"4\"><code>indented code</code></p
```
Note that the `li` element only contains `[ ] test` which matches the regex.

When `indented code` is indented with two or more spaces, the generated HTML looks like this:
```
<h2 class=\"raw part\" data-startline=\"1\" data-endline=\"1\">Header</h2>
<ul class=\"part\" data-startline=\"2\" data-endline=\"4\">
<li class=\"raw\" data-startline=\"2\" data-endline=\"4\">
<p>[ ] test</p>
<p><code>indented code</code></p>
</li>
</ul>
```
So the `li` element now has two `p`s inside. That confuses our code, as it only handles at most one `p` (it just removes it). The regex tries to match `\n<p>[ ] test</p>\n<p><code>indented code</code></p>\n`, which doesn't work.

**Comment by SISheogorath** [comment]
Specs seem to agree that this is how it's supposed to be rendered, so it's not an upstream bug:
https://spec.commonmark.org/0.29/#example-5

The question is weather we want to resolve this or not. I can imagine that it'll cause more problems when we make the filters less specific (i.e. replace every `[ ]` in a list element with a checkbox item).

Or should the checkbox item surround the entire content of the list element? Thoughts, ideas?

**Comment by xf-** [comment]
can you also match optional p infront? Only tag that can occur to cause this issue inside a list.
